src/KubernetesClient/Informers/ResourceObservableExtensions.cs

```diff
@@ -0,0 +1,299 @@
+                            .Where(x => x != null && !cacheSnapshot.ContainsKey(keySelector(x)))
+                            .Select(x => x.ToResourceEvent(EventTypeFlags.Add | EventTypeFlags.Computed))
+                            .ToList();
+                        var addedKeys = addedEntities
+                            .Select(x => keySelector(x.Value))
+                            .ToHashSet();
+
+                        var deletedEntities = cacheSnapshot
+                            .Where(x => !newKeys.Contains(x.Key))
+                            .Select(x => x.Value.ToResourceEvent(EventTypeFlags.Delete | EventTypeFlags.Computed))
+                            .ToList();
+                        var deletedKeys = deletedEntities
+                            .Select(x => keySelector(x.Value))
+                            .ToHashSet();
+
+                        // we can only compute updates if we are given a proper comparer to determine equality between objects
+                        // if not provided, will be sent downstream as just part of reset
+                        var updatedEntities = new List<ResourceEvent<TResource>>();
+                        if (comparer != null)
+                        {
+                            var previouslyKnownEntitiesInResetWindowKeys = cacheSnapshot
+                                .Keys
+                                .Intersect(resetBuffer.Select(x => keySelector(x.Value)));
+
+                            updatedEntities = resetBuffer
+                                .Where(x => previouslyKnownEntitiesInResetWindowKeys.Contains(keySelector(x.Value)))
+                                .Select(x => x.Value) // stuff in buffer that also existed in cache (by key)
+                                .Except(cacheSnapshot.Select(x => x.Value), comparer)
+                                .Select(x => x.ToResourceEvent(EventTypeFlags.Modify | EventTypeFlags.Computed))
+                                .ToList();
+                        }
+
+                        var updatedKeys = updatedEntities
+                            .Select(x => keySelector(x.Value))
+                            .ToHashSet();
+
+                        var resetEntities = resetBuffer
+                            .Select(x => x.Value)
+                            .Where(x => x != null &&
+                                        !addedKeys.Contains(keySelector(x)) &&
+                                        !deletedKeys.Contains(keySelector(x)) &&
+                                        !updatedKeys.Contains(keySelector(x)))
+                            .ToReset()
+                            .ToList();
+
+                        return deletedEntities
+                            .Union(addedEntities)
+                            .Union(updatedEntities)
+                            .Union(resetEntities);
+                    })
+                    .Do(_ => Console.WriteLine(_))
```

Comment:
It would be preferable if we didn't spam the console.

Reply:
Accidentally left when debugging (before I introduced logger). Will remove